server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -729,6 +740,34 @@ export class AgenticChatController implements ChatHandlers {
         return results
     }
 
+    /**
+     * Determines if error is thrown as a result of a user action (Ex. rejecting tool, stop button)
+     * @param err
+     * @returns
+     */
+    isUserAction(err: unknown, token?: CancellationToken): boolean {
+        return (
+            CancellationError.isUserCancelled(err) ||
+            err instanceof ToolApprovalException ||
+            (token?.isCancellationRequested ?? false)
+        )
+    }
+
+    async fetchModelResponse<RequestType, ResponseType>(
+        requestInput: RequestType,
+        makeRequest: (requestInput: RequestType) => Promise<ResponseType>
+    ): Promise<ResponseType> {
+        this.#debug(`Q Backend Request: ${JSON.stringify(requestInput)}`)
+        try {
+            const response = await makeRequest(requestInput)
+            this.#debug(`Q Backend Response: ${JSON.stringify(response)}`)
+            return response
+        } catch (e) {
+            this.#features.logging.error(`Error in call: ${JSON.stringify(e)}`)
+            throw new ModelServiceException(e as Error)
```

Comment:
Technically, we also throw `throw new Error('amazonQServiceManager is not initialized')` from within `generateAssistantResponse` (aka `makeRequest`) function. Which is something that should not be show to user.

Additionally, is exception thrown from Q API user-friendly enough to show it to user?

Reply:
We were asking the same question. We reached out the backend team and they said this should be okay for now, but we'll likely want something better in the long run. 

I was hoping to find a better solution here, but wasn't able to given the time constraints.